src/west/app/main.py

```diff
@@ -491,6 +512,24 @@ def run_command(self, argv, early_args):
         # If we're running an extension, instantiate it from its
         # spec and re-parse arguments before running.
 
+        if not early_args.help and early_args.command_name != "help":
+            # Recursively replace alias command(s) if set
+            aliases = self.aliases.copy()
+            while early_args.command_name in aliases:
+                new_args = shlex.split(aliases[early_args.command_name].args)
+                # Make sure we don't end up in an infinite loop
+                del aliases[early_args.command_name]
+                self.queued_io.append(lambda cmd: cmd.dbg(
+                    f'Replacing alias {early_args.command_name} with {new_args}'
```

Comment:
I'm testing the latest cead8d12a1d39f49ea88eaa733a2092ef9d992f2 and there is still a logging issue sorry:

```
west help

aliases:

  alpha:                beta
  beta:                 gamma
  gamma:                diff --stat

west -v alpha

Replacing alias diff with ['diff', '--stat']
Replacing alias diff with ['diff', '--stat']
Replacing alias diff with ['diff', '--stat']

```

Reply:
Thanks for spotting, fixed.

Reply:
Was this ONLY a logging bug? I could be wrong but the fix looks a big bigger than that...?

If and only if it was more than a logging bug, does this need more test coverage?

Reply:
It was a deferred logging issue, where the logging lambda didn't capture copies. To keep it clean, I've captured the alias itself (using `pop` vs `del`)

One addition here is that `shlex.split` is called when the aliases are created (these never change anyway).

I should have been more clear about that, sorry.